airflow-core/src/airflow/cli/commands/config_command.py

```diff
@@ -704,3 +705,148 @@ def lint_config(args) -> None:
+
+        2. Update only the 'core' and 'database' sections:
+            airflow config update --section core,database
+
+        3. Update only specific options:
+            airflow config update --option sql_alchemy_conn,dag_concurrency
+
+        4. Ignore updates for a specific section:
+            airflow config update --ignore-section webserver
+
+        5. Enable verbose output for more detailed information:
+            airflow config update --verbose
+
+    :param args: The CLI arguments for updating configuration.
+    """
+    console = AirflowConsole()
+    changes_applied = []
+
+    update_sections = args.section if args.section else None
+    update_options = args.option if args.option else None
+    ignore_sections = args.ignore_section if args.ignore_section else []
+    ignore_options = args.ignore_option if args.ignore_option else []
+
+    for change in CONFIGS_CHANGES:
+        conf_section = change.config.section
+        conf_option = change.config.option
+
+        if update_sections is not None and conf_section not in update_sections:
+            continue
+        if update_options is not None and conf_option not in update_options:
+            continue
+        if conf_section in ignore_sections or conf_option in ignore_options:
+            continue
+
+        if not conf.has_option(conf_section, conf_option, lookup_from_deprecated=False):
+            continue
+
+        current_value = conf.get(conf_section, conf_option)
+
+        if change.default_change and (str(current_value) != str(change.new_default)):
+            conf.set(conf_section, conf_option, str(change.new_default))
+            changes_applied.append(
+                f"Updated default value of '{conf_section}/{conf_option}' from '{current_value}' to '{change.new_default}'."
+            )
+
+        if change.renamed_to:
+            new_section = change.renamed_to.section
+            new_option = change.renamed_to.option
+            if not conf.has_section(new_section):
+                conf.add_section(new_section)
+            conf.set(new_section, new_option, current_value)
+            conf.remove_option(conf_section, conf_option)
+            changes_applied.append(f"Renamed '{conf_section}/{conf_option}' to '{new_section}/{new_option}'.")
+        elif change.was_removed:
+            conf.remove_option(conf_section, conf_option)
+            changes_applied.append(f"Removed '{conf_section}/{conf_option}' from configuration.")
+
+    for change in CONFIGS_CHANGES:
+        conf_section = change.config.section
+        conf_option = change.config.option
+
+        if update_sections is not None and conf_section not in update_sections:
+            continue
+        if update_options is not None and conf_option not in update_options:
+            continue
+        if conf_section in ignore_sections or conf_option in ignore_options:
+            continue
+
+        old_env = f"{ENV_VAR_PREFIX}{conf_section.upper()}__{conf_option.upper()}"
```

Comment:
Env vars need to be handled completely differently. All we can do is warn that we cannot do anything about them automatically. We can only help folks with things they have in their airflow.cfg.

Reply:
Should I skip checking the environment variables in this code and just focus on handling the airflow.cfg file? I feel thats better considering the purpose of this command

Reply:
Probably better to ignore them I think.